src/functor.jl

```diff
@@ -43,12 +43,11 @@ function _default_walk(f, x)
   re(map(f, func))
 end
 
-function fmap(f, x; exclude = isleaf, walk = _default_walk, cache = IdDict())
-  haskey(cache, x) && return cache[x]
-  y = exclude(x) ? f(x) : walk(x -> fmap(f, x, exclude = exclude, walk = walk, cache = cache), x)
-  cache[x] = y
+const INIT = Base._InitialValue()  # sentinel value for keyword not supplied
```

Comment:
Maybe less opaque to just define `struct NoPrune end`?

Reply:
Can do. Are you OK with the interface being that the *absence* of this keyword is how you turn this off? That's how `init` often works; and here, the values you might use for "don't prune" like `false` or `nothing` are all reasonable choices for the value to use during pruning.

Reply:
The code is weird to me, but the API makes sense. I'm happy.